ledger/participant-integration-api/src/main/scala/platform/store/utils/ConcurrencyLimiter.scala

```diff
@@ -49,7 +56,15 @@ class QueueBasedConcurrencyLimiter(
     while (running < parallelism && waiting.nonEmpty) {
       val head = waiting.dequeue()
       running = running + 1
-      head()
+      parentO match {
+        case Some(parent) =>
+          // defering creating the tasks's future to the parent concurrency limiter
+          parent.execute(head())
```

Comment:
this is the right ordering. blocked parent starves the child pool ("running" futures doing nothing). The other way around would be bad.

Reply:
in practice, without these changes it should go like: locallimitter.execute(globallimitter.execute(stuff))

Reply:
Is your recommendation to cancel the changes from this PR and instead compose then at call sites like in your example?

Reply:
I could see value in defining some combination function on the companion for the trait, which does the right order in composing limiters, and adds tests for them, that 
* global limits are respected
* global limiter is not flooded by local limiting waiting tasks